core/app/components/pubs/PubEditor/PubEditor.tsx

```diff
@@ -269,12 +272,17 @@ export async function PubEditor(props: PubEditorProps) {
 	const currentStageId = pub?.stage?.id ?? ("stageId" in props ? props.stageId : undefined);
 	const pubForForm = pub ?? { id: pubId, values: [], pubTypeId: form.pubTypeId };
 
+	// For the Context, we want both the pubs from the initial pub query (which is limited)
+	// as well as the pubs related to this pub
+	const relatedPubs = pub ? pub.values.flatMap((v) => (v.relatedPub ? [v.relatedPub] : [])) : [];
```

Comment:
TIL: you can use `flatMap` to get around how doing a `.map(...).filter(value => !!value)` doesn't let TS think you have `Thing[]` and TS will still think you have `(Thing | null)[]` even after the filter

https://stackoverflow.com/a/59726888

Reply:
very hot tip, the .filter typing is so annoying!

Reply:
i think they did fix this in a recent release, if you do `.filter(value=>value != null)` specifically!

Reply:
oh no nvm, this only applies one level deep! they fixed the "simple" case in typescript 5.5

https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#inferred-type-predicates

Example (change to typescript <5.5 to see error in the simple case)

https://www.typescriptlang.org/play/?target=99&pretty=true&allowSyntheticDefaultImports=true&preserveValueImports=true&isolatedModules=true&verbatimModuleSyntax=false&ts=5.5.4&ssl=1&ssc=1&pln=2&pc=1#code/C4TwDgpgBAwg9gWzAGwgDwCoAsCWA7AcygF4oBvKLCEALnMrjoHIrlk4omBfKAHyjwBXNlwBQoSFADKOJKmz4ipClVqdW7JlB78hbUQBMIAY2QBDAE7RjcPAGdgUG3PR0AFPBeZchAJQBtAF1DE3MrJ1sHKDtZFAg6GRcFQiDRURt7R2cwEgivADoAMxxkYAgLNwA3YgA+SvzVKABCUj1kX1FURys7XOz-AAZAhuoGjlEAegmoGagAPQB+KDSMqJikXPW4opKyiuqapqbKjq6oHoAmTdjB4dVJ6dnFqCA